### DISCUSSION TITLE
Simulate mouse events programmatically

### DISCUSSION THREAD

**Discussion by vkaelin** [matteobruni/tsparticles#4907] [Q&A] [answered]
Hello,

Sorry to bother you again but I'm trying to simulate mouse events programmatically (I want to fake a "click" from the user at a specific location on the screen).

I saw your answer on this question: https://github.com/matteobruni/tsparticles/discussions/4879#discussioncomment-4585619 and I tried to implement it in the following codepen:

https://codepen.io/Kalane/pen/GRBVVwq

As you can see in the console, the `clickHandler` is receiving the event but sadly nothing is displayed on the screen.

I also tried to directly add the particle to the state (in the `container.actualOptions.manualParticles` object) but it didn't do anything.

Thanks a lot!

**Comment by matteobruni** [comment, marked as answer]
This examples simulates mouse events: https://codepen.io/matteobruni/pen/ExPoybM

Anyway, what is the expected result? Random confetti explosions can be made without all this effort, the animation can be started when needed. If you need specific positions let me know, I can create a sample adding emitters when needed

**Reply by vkaelin** [reply]
Hello again,

Yes thanks that's the codepen I tried to edit to make it work but without success for now.

Here is my use case: I want to generate particles on specific locations when I receive events from my backend (websockets). 

I don't want random positions as every event from the server will send the precise location of the desired particles.

Thanks a lot again for your time and this very cool lib!

**Reply by matteobruni** [reply]
Checkout this sample: https://codepen.io/matteobruni/pen/qBMWqdW?editors=1111

It uses the `addEmitter` function, provided by the `tsparticles-plugin-emitters` (you should already have it loaded). The first parameter is the emitters options, the second one is the position in pixels. Be careful of retina displays, `container.retina.pixelRatio` contains the value to multiply it if you return non retina coordinates.

I think this is the best solution for you case.

**Reply by vkaelin** [reply]
Thanks! It works great!
